src/snowflake/connector/connection.py

```diff
@@ -1541,3 +1555,22 @@ def _all_async_queries_finished(self) -> bool:
             not self.is_still_running(self.get_query_status(q)) for q in queries
         )
         return all(finished_async_queries)
+
+    def _log_telemetry_imported_packages(self) -> None:
+        if self._log_imported_packages_in_telemetry:
+            # filter out duplicates caused by submodules
+            # and internal modules with names starting with an underscore
+            imported_modules = {
+                k.split(".", maxsplit=1)[0]
```

Comment:
It's possible that several packages would have the same top level module, e.g. `snowflake`. But this can be considered as corner cases. There probably isn't a one-size-fit-all solution.

Reply:
yea, that's possible but I think these packages are rare. Otherwise the telemetry will be blown up by submodules (simply import numpy you will get ~30 submodules)